pagefaultlatency/pagefaultlatency.bt

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/bpftrace
+
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+/*
+ * pagefaultlatency.bt    Pagefault latency tracing tool.
+ *
+ * USAGE: wakesnoop.bt [cmd]
+ */
+
+BEGIN
+{
+    if ($# == 1) {
+        @proc = str($1);
+        printf("Tracing pagefault latency for comm[%s]\n", @proc);
+    } else {
+        printf("No command specified!\n");
+        exit();
+    }
+}
+
+kprobe:handle_mm_fault
+/comm == @proc/
+{
+    @start[tid] = nsecs();
+}
+
+kretprobe:handle_mm_fault
+/@start[tid]/
+{
+    @usecs = hist((nsecs - @start[tid]) / 1000);
+    @latency = nsecs() - @start[tid];
```

Comment:
Latency doesn't seem to live outside of this function. It could be just a scratch variable (which is much faster).